### DISCUSSION TITLE
An IoT course based on MicroPython

### DISCUSSION THREAD

**Discussion by uraich** [micropython/micropython#9425] [General]
Since 3 years I am developing a course on IoT for several African universities. The course is based on the WeMos D1 mini bus with an ESP32 as microcontroller and MicroPython as programming language. The course is described on a [TWiki server](https://afnog.iotworkshop.africa/do/view) and is available in slightly different versions in English and French (see Site Map). The solutions to the exercises are available on [github](https://github.com/uraich/IoT-Course). You will find quite a few pure Python drivers for the different sensor shields available for the WeMos D1.  Some other sensors were interfaced through prototype boards and drivers for these are also available. Maybe someone will find there the driver he needs for his project.

**Comment by robert-hh** [comment]
This is a really good course, helpful for all beginners who need a systematic introduction. Especially the exercises section is very good. I know from helping my children, that finding good exercises is a hard task of finding the right balance.
